src/libraries/Microsoft.PowerFx.Connectors/Execution/IConnectorInvoker.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors.Execution
+{
+    public interface IConnectorInvoker
+    {
+        // runtime context
+        BaseRuntimeConnectorContext Context { get; }
+
+        // request: classic call
+        Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancellationToken);
+
+        // resquest: support for paging
+        Task<FormulaValue> InvokeAsync(string nextlink, CancellationToken cancellationToken);        
+        
+        // processing/inner call
+        Task<FormulaValue> SendAsync(InvokerParameters invokerElements, CancellationToken cancellationToken);
```

Comment:
>SendAsync [](http://example.com/codeflow?start=27&length=9)

Why do we need both InvokeAsync and SendAsync? Is one layered on the other?

Reply:
InvokeAsync is high level and transforms Fx parameters to InvokerParameters. This is where we apply the "policy", aligning the input parameters with what the swagger file mandates.

SendAsync transforms InvokerParameters to a result (FormulaValue) and that's the only abstract method in FunctionInvoker.

It is implemented in HttpFunctionInvoker (where it's the only API being implemented, generating an HttpRequestMessage, getting an HttpResponse and transforming it to a FormulaValue)

This is the key part that allows us to have a custom invoker
You see this in CustomInvokerTests.cs, in CustomFunctionInvoker class, which is making a call to CustomInvoker, itself an equivalent of HttpClient

Reply:
Notice that this approach could help us in our tests, in not using an HttpClient